### ISSUE TITLE
Github Desktop Opening Inspect Element Instead of App

### ISSUE THREAD

**Issue by HFocus** [desktop/desktop#3107]
### Description

Downloaded Github Desktop for the first time on this computer. When I open the app by any means, a blank html editor like chrome's inspect element appears instead of the app.

### Version

**GitHub Desktop version:** 1.0.4

**OS version:** 10.0.15063

### Steps to Reproduce

1. Download Github Desktop 1.0.4 using Chrome
2. Click on open when done
3. Github will open itself when done installing displaying blank html document

![image](https://user-images.githubusercontent.com/31972815/31799537-e0ebaf16-b4f7-11e7-85e6-03faf4716e1d.png)

**Expected behavior:** Initial opening of Github Desktop

**Actual behavior:** Opening of the blank document (task name is still Desktop)

**Reproduces how often:** 100%

### Logs

No logs folder found in AppData\Roaming\GitHub Desktop\

**Comment by shiftkey** [comment]
@HFocus that's interesting - can you switch over to the Console tab and see what that error is that it's reporting?

**Comment by HFocus** [comment]
Not allowed to load local resource: file:///C:/Users/The%20Kong%20#2\AppData\Local\GitHubDesktop\app-1.0.4\resources\app/index.html

**Comment by j-f1** [comment]
Huh. It looks like the `#` in your username is being interpreted as the start of a URL hash, and should probably be escaped.

**Comment by shiftkey** [comment]
@HFocus yep, what @j-f1 said. I'm pretty sure it's [this line](https://github.com/desktop/desktop/blob/master/app/src/main-process/app-window.ts#L136) that needs to be formatted correctly:

https://github.com/desktop/desktop/blob/a6fcd1a3feb424fae03bc01d0909d75ec1a0d7e3/app/src/main-process/app-window.ts#L136

**Comment by j-f1** [comment]
Looks like we’re missing an `encodeURIComponent`.

**Comment by shiftkey** [comment]
@j-f1 it looks like there's _some_ encoding (spaces -> `%20`)  and also the mix of `\` and `/` is worrying - this might need some more research after a hotfix.

**Comment by j-f1** [comment]
> mix of `\\` and `/` is worrying

I’m pretty sure that `\\` and `/` are equivalent in URLs. The only reason the path after the `#` uses `\\` is because it isn’t translated since it’s part of the hash.

> it looks like there's some encoding

I wonder if that’s error correction by Chrome — spaces are *never* valid in a URL, so they can always be replaced with `%20`.

**Comment by shiftkey** [comment]
> I’m pretty sure that \\ and / are equivalent in URLs. 

While they might be equivalent, this suggests two different sources of resolving these paths:

 - `C:/Users/The%20Kong%20#2` - most Windows environment variables use `\\` here, which suggests it's not using those
 - `\AppData\Local\GitHubDesktop\app-1.0.4\resources\app` - now we're into the app resources, and they're encoded with a `\\`
 - `/index.html` - this is the problem line

That's what I want to get to the bottom of.

> I wonder if that’s error correction by Chrome — spaces are never valid in a URL, so they can always be replaced with %20.

I'm not sure. Electron is responsible for setting `__dirname` and `__filename` as globals in the renderer, so I want to see where they're getting those values from, and if they're doing any encoding in advance.

@HFocus I've been able to reproduce this using an account containing a `#`

**Comment by j-f1** [comment]
> That's what I want to get to the bottom of.

`file://${encodeURIComponent(Path.join(__dirname, 'index.html'))}` should fix these issues. (for some environment where `Path` is the builtin `path` module)

**Comment by shiftkey** [comment]
We use `__dirname` in a lot of other places. I don't want to rush out a fix for this issue and then encounter a bunch of other things down the track.

**Comment by shiftkey** [comment]
@HFocus could you open a PowerShell prompt and run this command?

```
$ Get-ChildItem Env:
```

I have a theory that the root of this issue is in [this part](https://github.com/electron/electron/blob/77d4927e8bcbb7a80febf2d3424e3129f50ceb9e/lib/renderer/init.js#L111-L130) of Electron's setup, but it requires fiddling with the initialisation of Electron and I want to ensure my environment matches yours.

**Comment by HFocus** [comment]
![image](https://user-images.githubusercontent.com/31972815/31869220-c86fe6be-b763-11e7-9942-19c8a290eb85.png)

**Comment by shiftkey** [comment]
@HFocus thanks for the details. I think #3109 will fix this, and I won't hold up that PR understanding why the `%USERPROFILE%` path is formatted differently for your machine - we handle this already in that PR.
